### PR TITLE
Sales logger

### PR DESCRIPTION
**Pull request by darmhoo**
#### What does this PR do?
- (replace me) Explain WHY this PR is required. E.g.: What feature it adds, what problem it solves, …

#### Description of Task proposed in this pull request?
- (replace me) A description of WHAT changed in the codebase. Additionally add the reasoning if it's complex or abstract. 

#### How should this be manually tested (Quality Assurance)?
- Explain how to manually test this feature. For example if changes were made in
the UI or in the API, explain where and if any specific access is needed.

#### What are the relevant pivotal tracker stories?
- [Story ID:  replace text with story id](paste Pivotal Tracker link here)

#### Any background context you want to add (Operations Impact)?
- (replace me) List the steps that must be taken for this PR to work.
E.g.: `npm install`, `Add "github_key" to environment variables, ...`

Be sure to include all systems that needs to be changed or which system is
affected by the change (Ex: Requires Styled Components to be installed and
configured in `babel.rc`, file to be updated).

#### What I have learned working on this feature: [If you don't put anything here, you are doing it wrong!]
- (replace me) I learned how to create a pull request template

### CODE REVIEW

File: resources/assets/js/views/FSL/cash_logger/cash_logger.vue [hunk 1]
```diff
@@ -27,189 +27,207 @@
...
-                        </div>
 
-                        <div class="tab-content mt-1 attendance-body">
-                            <div class="tab-pane active text-center" v-if="show && customer.orders.length > 0">
-                                <div class="mb-3 row attendance-item" v-for="(order, index) in customer.orders">
-                                    <div class="col-12 col-xs-2 col-md col-lg d-flex align-items-center"
-                                         style="max-width: 100px">
-                                        <span class="user mx-auto">{{index+1}}</span>
-                                        <span v-if="$route.meta.customSMS">
+                            <div class="tab-content mt-1 attendance-body">
```
Comment (jhayX) [inline resources/assets/js/views/FSL/cash_logger/cash_logger.vue:55]: Can this be re used in the third screen?
Please follow the mockup design
Also when you click on the comment  the full comment isn't revealed.

File: resources/assets/js/views/FSL/cash_logger/cash_logger.vue [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+                        </a>
+                    </li>
+                </ul>
+            </div>
+            <div class="tab-content py-3">
+                <div class="tab-pane fade" :class="{ 'active show': isActive('home') }" id="home">
+                    <AutoCompleteSearch
+                        title=""
+                        @customer-selected="processForm"
+                        :url="'/api/customer/autocomplete'"/>
```
Comment (AltaraNg) [inline resources/assets/js/views/FSL/cash_logger/cash_logger.vue:41]: Going forward, lets have route helper, this shouldnt be inside a code

File: resources/assets/js/views/FSL/cash_logger/cash_logger.vue [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+                this.$LIPS(false);
+            },
+
+            submitForm() {
+                this.cashLogForm.customer_id = this.customer.id;
+                this.$validator.validateAll().then(result => {
+                    if (result) {
+                        if (this.$network()) {
+                            this.$LIPS(true);
+                            post('/api/payment', this.cashLogForm).then((res) => {
```
Comment (AltaraNg) [inline resources/assets/js/views/FSL/cash_logger/cash_logger.vue:302]: Let's handle our backend calls with services or store, to have a testable component and reduce side effect

File: resources/assets/js/views/FSL/cash_logger/cash_logger.vue [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+
+            logCashLog(){
+                let desc = `${this.user.name} logged ${$formatCurrency(this.cashLogForm.amount)} of customer: ${this.customer.id}`;
+                return log('Cash Logged', desc);
+            },
+
+            processForm(id) {
+                this.show = false;
+                this.$LIPS(true);
+                get(`/api/customer/lookup/${id}`)
```
Comment (AltaraNg) [inline resources/assets/js/views/FSL/cash_logger/cash_logger.vue:324]: Same here

File: resources/assets/js/views/FSL/cash_logger/cash_logger.vue [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+                        this.currentPage -= 1;
+                        break;
+                    default:
+                        this.currentPage = value
+                }
+                this.query.page = this.currentPage;
+                this.$LIPS(true);
+
+                get(
+                    `/api/payment?page=` +
```
Comment (AltaraNg) [inline resources/assets/js/views/FSL/cash_logger/cash_logger.vue:360]: Also here

File: resources/assets/js/views/FSL/cash_logger/cash_logger.vue [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+            this.$prepareBranches();
+            this.$preparePaymentMethods();
+
+        },
+       async mounted(){
+            try {
+                if (this.$network()) {
+                    this.$LIPS(true);
+                    this.error = {};
+                    get('/api/payment-type').then(res => {
```
Comment (AltaraNg) [inline resources/assets/js/views/FSL/cash_logger/cash_logger.vue:380]: this

File: resources/assets/js/views/FSL/cash_logger/cash_logger.vue [hunk 7]
```diff
@@ -0,0 +1,415 @@
...
+                        Vue.set(this.$data, 'paymentType', res.data);
+                    }).catch(() => {
+                        Flash.setError("Unable to fetch payment type")
+                    });
+                    get('/api/payment').then(res => {
```
Comment (AltaraNg) [inline resources/assets/js/views/FSL/cash_logger/cash_logger.vue:385]: this

File: resources/assets/js/views/FSL/cash_logger/cash_logger.vue [hunk 8]
```diff
@@ -0,0 +1,392 @@
+<template>
+    <transition name="fade">
+
+        <div id="cash-logger">
+
+            <div class="mt-5 mb-3 attendance-head">
+                <ul class="nav nav-tabs justify-content-center p-0" role="tablist">
```
[the remark this answers is not in the document]
Comment (darmhoo) [reply resources/assets/js/views/FSL/cash_logger/cash_logger.vue:7]: Hi. I don't get.

**Pull request merged**